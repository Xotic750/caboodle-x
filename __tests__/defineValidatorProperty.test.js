import 'babel-polyfill';
import {defineValidatorProperty} from '../dist/caboodle-x';

const has = Object.prototype.hasOwnProperty;
const doc = typeof document !== 'undefined' && document;
const itHasDoc = doc ? it : xit;

describe('defineValidatorProperty', () => {
  let obj;

  beforeEach(() => {
    obj = {};

    defineValidatorProperty(obj, 'name', {
      configurable: true,
      enumerable: true,
      nilable: false,
      validator: () => true,
      value: 'Testing',
      writable: true,
    });
  });

  it('should return the initial value', () => {
    expect(has.call(obj, 'name')).toBe(true);
    expect(obj.name).toBe('Testing');
  });

  it('should be setable', () => {
    obj.name = 'Other';
    expect(obj.name).toBe('Other');
  });

  it('should return the parent initial value', () => {
    const child = Object.create(obj, {});

    expect(child.name).toBe('Testing');
    expect(has.call(child, 'name')).toBe(false);
  });

  it('should not override the parent value', () => {
    const child = Object.create(obj, {});

    defineValidatorProperty(child, 'name', {value: 'Other'});

    expect(obj.name).toBe('Testing');
    expect(child.name).toBe('Other');
  });

  it('should throw error for non object', () => {
    expect(() => {
      defineValidatorProperty(42, 'name', {});
    }).toThrowErrorMatchingSnapshot();
  });

  it('should not throw error for empty descriptor', () => {
    defineValidatorProperty({}, 'name', {});
  });

  itHasDoc('works with DOM elements', () => {
    const div = document.createElement('div');
    defineValidatorProperty(div, 'blah', {value: 1});
    expect(div.blah).toBe(1);
  });
});
