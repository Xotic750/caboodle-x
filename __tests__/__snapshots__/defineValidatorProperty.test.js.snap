// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defineValidatorProperty should throw error for non object 1`] = `"Not an object."`;
